src/utils/usageReportConsent.js

```diff
@@ -1,13 +1,14 @@
-const KEY = 'usage-data-consent'
+import { appVersion } from 'utils'
 
-export const getConsentStatus = () => {
-  return (localStorage.getItem(KEY) === 'true')
-}
+const KEY = 'usage-data-consent'
 
-export const setConsentStatus = consent => {
-  localStorage.setItem(KEY, !!consent)
+export const isConsentGranted = () => {
+  return localStorage.getItem(KEY) !== null
 }
 
-export const hasConsentBeenAnswered = () => {
-  return (localStorage.getItem(KEY) !== null)
+export const grantConsent = () => {
+  localStorage.setItem(KEY, JSON.stringify({
+    timestamp: Date.now(),
+    version: appVersion()
```

Comment:
we set the version but never check it, do we need to show the consent every version update?

Reply:
There is a chance we need to send the consent to the server eventually. Setting the timestamp and version makes it very easy for us to deal with consent granted on an older version of the app.